doc/designs/fast-param-find.md

```diff
@@ -0,0 +1,80 @@
+Format:
+
+```
+    %% FUNCTION-BASE-NAME
+
+    prologue
+
+    % PARAM-NAME type [flags]
+    code to process param
+
+    % PARAM-NAME type [flags]
+    code to process param
+
+    %
+
+    epilogue
+```
+
+- the % characters must be start of line
+
+- FUNCTION-BASE-NAME is the name the functions will derive from e.g. "kdf_scrypt" to this will be appended _[gs]et[_ctx]_params
+
+- type is the standard OSSL_PARAM types with additions for basic C types
+
+- Flags include:
+  - alias (uses same name as another entry but different type)
+  - multiple (multiple instances are allowed - not sure if necessary)
```

Comment:
We had this for the multiple CRT params in RSA keys, but we changed that to have indexed names instead.  I can't remember any other case where we did this.

Reply:
We concatenate the KDF INFO params.

Reply:
But maybe we could just say that you need to call the set param multiple times if you want this.

Reply:
I'm not worried about the concatenation problem without calling set param multiple times.
I'm not even sure we actually need the _multiple_ flag.

Reply:
More important is the general outline of the suggestion not the implementation specifics.
No rabbit holes.